src/app/qgsmeasuredialog.cpp

```diff
@@ -238,14 +238,33 @@ void QgsMeasureDialog::mouseMove( const QgsPointXY &point )
   {
     QVector<QgsPointXY> tmpPoints = mTool->points();
     tmpPoints.append( point );
-    const double area = mDa.measurePolygon( tmpPoints );
+    double area = 0;
+    try
+    {
+      area = mDa.measurePolygon( tmpPoints );
+    }
+    catch ( QgsCsException & )
+    {
+      // TODO report errors to user
```

Comment:
Can we get rid of some (all of) those TODOs? here for instance, we have access to QgisApp visibleMessageBar()

Reply:
I'm happy for someone else to do that, but right now it's not my priority. There's no regression here at least, the behaviour will be identical to current behaviour